goldmane/pkg/aggregator/bucketing/bucket_ring.go

```diff
@@ -25,7 +25,27 @@ import (
 )
 
 type StreamReceiver interface {
-	ReceiveFlow(*types.DiachronicFlow, int64, int64)
+	Receive(FlowBuilder)
+}
+
+type FlowBuilder interface {
```

Comment:
Could you add a comment or two about the purpose of this? Just stating something like it's used so we don't take up memory building the flow until we need to?

Reply:
I actually really like this though, I think it's a good practice of not consuming more memory than then needed until we actually need it.